From a2beb9aa105d23066f32d7a16923432bfe39a4a4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Kongstad?= <amhk@google.com>
Date: Fri, 20 Feb 2026 15:20:37 +0000
Subject: [PATCH] Fix errorprone build after SDK ext 22

SDK extensions version 22 introduced
isAutoJoinInAdvancedProtectionModeEnabled, and now errorprone breaks
because of

  [...]/StandardWifiEntry.java:629: \
       Error: Call requires API level CUR_DEVELOPMENT/10000 (current min is 31): \
       android.net.wifi.WifiConfiguration#isAutoJoinInAdvancedProtectionModeEnabled [NewApi]

This is a false positive, and the access to
isAutoJoinInAdvancedProtectionModeEnabled is correctly flag checked. Add
@SuppressLint to unblock the API finalization.

Bug: 481594013
Test: lunch sdk-trunk_staging-userdebug && m out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/lint-baseline.xml out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/lint-report.html out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/lint-report.txt out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/lint-report.xml out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/lint-srcs.list out/soong/.intermediates/frameworks/opt/net/wifi/libs/WifiTrackerLib/SdkWifiTrackerLib/android_common/lint/suggested-fixes.zip
Flag: NONE platform SDK finalization
Change-Id: I664d859e4dca87beed4e15069b88ec4dd4c48c2b
---
 .../src/com/android/wifitrackerlib/StandardWifiEntry.java        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java b/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
index 6bcb06cf838..9b4943daeb3 100644
--- a/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
+++ b/libs/WifiTrackerLib/src/com/android/wifitrackerlib/StandardWifiEntry.java
@@ -621,6 +621,7 @@ public class StandardWifiEntry extends WifiEntry {
     }
 
     @Override
+    @SuppressLint("NewApi")
     public synchronized boolean isAutoJoinEnabled() {
         if (mTargetWifiConfig == null) return false;
 
-- 
2.53.0.345.g96ddfc5eaa-goog

